processing/src/main/java/io/druid/segment/DimensionSelector.java

```diff
@@ -150,12 +150,24 @@ default long getLong()
   }
 
   /**
+   * @deprecated always throws {@link UnsupportedOperationException}
+   */
+  @Deprecated
+  @Override
+  default boolean isNull()
+  {
+    throw new UnsupportedOperationException("DimensionSelector cannot be operated as numeric ColumnValueSelector" + this.getClass());
```

Comment:
`DimensionSelector` is closer to be an "object" selector, then numeric. So I think this method should just return false. Indeed, it couldn't return `null` from getLong/Double/Float, but it *could* return null from it's getObject().

Reply:
done.